cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionConsumerResult.java

```diff
@@ -24,9 +24,10 @@
  */
 public class PartitionConsumerResult {
   private final PartitionConsumerState partitionConsumerState;
-  private final Iterator<Partition> partitionIterator;
+  private final List<PartitionDetail> partitionIterator;
```

Comment:
Now that a partition can expire (and is then considered as failed), should this result object also return the set of partition keys that expired?